spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/pubsub/GcpPubSubAutoConfiguration.java

```diff
@@ -128,26 +135,67 @@ public PubSubTemplate pubSubTemplate(PublisherFactory publisherFactory,
 	@ConditionalOnMissingBean
 	public SubscriberFactory defaultSubscriberFactory(
 			@Qualifier("subscriberExecutorProvider") ExecutorProvider executorProvider,
+			@Qualifier("subscriberSystemExecutorProvider")
+			@Nullable ExecutorProvider systemExecutorProvider,
+			@Qualifier("subscriberFlowControlSettings") @Nullable FlowControlSettings flowControlSettings,
```

Comment:
Can we get rid of these `@Nullable` and maybe replace them by `@Autowired(required = false)` and Mike suggested?

Reply:
What's the difference?

Reply:
I would say more Spring idiomatic?

Reply:
No `@Autowired` in the auto-configurations, please.
There is something like `ObjectProvider` in Spring Boot to consider instead.

Reply:
@artembilan what's the best practice in this situation? Is `@Nullable` commonly used?
@ChengyuanZhao If so, can we do a post-merge fix?

Reply:
The practice is to use an `ObjectProvider<?>`.
See here fore example: https://github.com/spring-projects/spring-boot/blob/master/spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/jdbc/DataSourceJmxConfiguration.java#L58

Reply:
I can make the change to use the objectprovider in the next PR.

Reply:
Correct, because this one is merged already.